### PR TITLE
Deprecating wit_var to the benefit of its synonymous wit_hyp

### PR DESCRIPTION
**Pull request by herbelin**
**Kind:** simplification

This supposingly goes towards a better lisibility of what to use for plugins developers.

Notice that `hyp` was documented in the `Tactic Notation` section of the reference manual but `var` was not. Notations using the undocumented `var` (such as `Tactic Notation "f" var(id) := clear id.`) need to (equivalently) use `hyp`.

- [X] dev/doc/changes.md updated

### CODE REVIEW

**Comment by ejgallego** [comment]
Related to #9122

**Comment by herbelin** [comment]
> Related to #9122

Related and apparently complementary!

Do you want help on #9122 (e.g. for overlays, or on the setting of a deprecation phase)?

**Review by ejgallego — APPROVED** [review]
LGTM, however this seems more to be a documentation issue so I'll leave @coq/ltac-maintainers @coq/doc-maintainers to handle this PR.

**Comment by ejgallego** [comment]
> Related and apparently complementary!
> 
> Do you want help on #9122 (e.g. for overlays, or on the setting of a deprecation phase)?

Indeed it is complementary; I'd say don't worry too much about it @herbelin , of course if you want to rebase it help is much appreciated, but I should if not get some cycles myself to tweak it some of these days.

File: parsing/g_prim.mlg
Comment (jfehrle) [inline parsing/g_prim.mlg:69]: Maybe slide up above identref, next to the deletion.

File: plugins/ltac/extratactics.mlg
Comment (jfehrle) [inline plugins/ltac/extratactics.mlg:451]: Doesn't subst take variables rather than hypotheses?  I don't think we'd want the doc to say "hyp" here if it's not a hypothesis.  At the moment, the edited doc grammar uses `ident` here, but we've not applied those changes in the the tactics chapter yet.

Maybe should make hyp and var synonyms without deprecating?

File: plugins/ltac/tacentries.ml
Comment (jfehrle) [inline plugins/ltac/tacentries.ml:223]: Ah, there is a warning.

**Review by jfehrle — COMMENTED** [review]
I'm all for simplification, but a couple questions here:

**Review by gares — APPROVED** [review]
ssr ok

File: parsing/g_prim.mlg
Comment (herbelin) [reply parsing/g_prim.mlg:69]: I moved it because it now depends on `identref`. This part of code will anyway be reorganized in #11842.

File: plugins/ltac/extratactics.mlg
Comment (herbelin) [reply plugins/ltac/extratactics.mlg:451]: That's actually the big question! Internally, definition=theorem, body=proof, variable=hypothesis, parameter=axiom, recursion=induction, ... that's the whole issue of documenting a unique syntax which has a double meaning. This leads to a tension between providing intuitive names and avoiding redundancy and this is a major issue in the pedagogy of Coq.

I'm afraid that if we really want to maintain a difference of vocabulary, the difference would have to be kept in the structure of terms themselves, so that this difference (decided by the user) is consistently used for parsing and printing. Why not? But on the other side, I tend to think that users, after some while, would themselves get use to the identification of concepts and stop making a formal difference (as most Coq users do after some time, where we freely switch between the programming vocabulary and the logic vocabulary, meaning either of them).

So, shortly, this is indeed a critical pedagogical issue.

File: interp/stdarg.mli
Comment (herbelin) [reply interp/stdarg.mli:43]: I copied what was done in other deprecation messages. I agree that in this case, it should be obvious that this is in the same module. On the other side, I guess OCaml programmers know if and when they want to qualify a name.

I may propose the following alternative: `"Use wit_hyp from module Stdarg"`.

**Comment by Zimmi48** [comment]
Who's assigning?

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
Merging soon.

**Comment by ppedrot** [comment]
@herbelin this needs a rebase, I'll merge it after assuming CI is fine.

**Pull request merged**
